Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Generator.cs

```diff
@@ -85,16 +91,38 @@ public void Execute(GeneratorExecutionContext context)
                 }
 
                 var isExecutable = false;
+
+                bool foundFatalError = false;
                 
                 var assemblyAttributes = context.Compilation.Assembly.GetAttributes();
                 
                 // Let's find the AssemblyTitleAttribute
                 var generateMainAttribute = assemblyAttributes
-                    .FirstOrDefault(attr => attr.AttributeClass.Name == nameof(LambdaGenerateMainAttribute));
+                    .FirstOrDefault(attr => attr.AttributeClass.Name == nameof(LambdaGlobalPropertiesAttribute));
+
+                var defaultRuntime = "dotnet6";
 
                 if (generateMainAttribute != null)
                 {
-                    isExecutable = true;
+                    var generateMain = generateMainAttribute.NamedArguments.FirstOrDefault(kvp => kvp.Key == "GenerateMain").Value;
+                    var runtime = generateMainAttribute.NamedArguments.FirstOrDefault(kvp => kvp.Key == "Runtime")
+                        .Value.Value.ToString();
+
+                    if (!_allowdRuntimeValues.Contains(runtime))
+                    {
+                        diagnosticReporter.Report(Diagnostic.Create(DiagnosticDescriptors.InvalidRuntimeSelection, Location.None));
+                        return;
+                    }
+
+                    defaultRuntime = runtime;
+                    
+                    isExecutable = bool.Parse(generateMain.Value.ToString());
+                    
+                    // if (isExecutable && context.Compilation.Options.OutputKind != OutputKind.ConsoleApplication)
```

Comment:
Why is this code commented out? Seems like we need it.

Reply:
@normj I wanted to have a chat about it, I meant to message you. When enabled, all the tests fail. I think it's because the test framework is not an executable, the OutputKind is always set to DLL. Regardless what options are set in the csproj file.

Reply:
I was able to get the `VerifyExecutableAssembly` test to pass when I uncommented this code and in the test I set the `OutputKind` property on the `TestState`  and when you are including the sources of `Amazon.Lambda.RuntimeSupport` exclude the `Program.cs` from that project. Below is my updated version of the test. I think you can do the same think to the other executable tests that are failing.

```
[Fact]
public async Task VerifyExecutableAssembly()
{
    var expectedTemplateContent = (await File.ReadAllTextAsync(Path.Combine("Snapshots", "ServerlessTemplates", "subnamespace_executableimage.template"))).ToEnvironmentLineEndings();
    var expectedSubNamespaceGenerated = (await File.ReadAllTextAsync(Path.Combine("Snapshots", "Functions_AsyncStartupToUpper_Generated.g.cs"))).ToEnvironmentLineEndings();
    var expectedProgramGenerated = (await File.ReadAllTextAsync(Path.Combine("Snapshots", "Program.g.cs"))).ToEnvironmentLineEndings();
            
    var test = new VerifyCS.Test
    {
        TestState =
        {
            OutputKind = OutputKind.ConsoleApplication,
            Sources =
            {
                (Path.Combine("TestExecutableServerlessApp", "Sub1", "Functions.cs"), await File.ReadAllTextAsync(Path.Combine("TestExecutableServerlessApp", "Sub1", "Functions.cs"))),
                (Path.Combine("TestExecutableServerlessApp", "Startup.cs"), await File.ReadAllTextAsync(Path.Combine("TestExecutableServerlessApp", "Startup.cs"))),
                (Path.Combine("TestExecutableServerlessApp", "Services", "SimpleCalculatorService.cs"), await File.ReadAllTextAsync(Path.Combine("TestExecutableServerlessApp", "Services", "SimpleCalculatorService.cs"))),
                (Path.Combine("Amazon.Lambda.Annotations", "LambdaFunctionAttribute.cs"), await File.ReadAllTextAsync(Path.Combine("Amazon.Lambda.Annotations", "LambdaFunctionAttribute.cs"))),
                (Path.Combine("Amazon.Lambda.Annotations", "LambdaStartupAttribute.cs"), await File.ReadAllTextAsync(Path.Combine("Amazon.Lambda.Annotations", "LambdaStartupAttribute.cs"))),
                (Path.Combine("Amazon.Lambda.Annotations", "LambdaGlobalPropertiesAttribute.cs"), await File.ReadAllTextAsync(Path.Combine("Amazon.Lambda.Annotations", "LambdaGlobalPropertiesAttribute.cs"))),
                (Path.Combine("TestExecutableServerlessApp", "AssemblyAttributes.cs"), await File.ReadAllTextAsync(Path.Combine("TestExecutableServerlessApp", "AssemblyAttributes.cs"))),
            },
            GeneratedSources =
            {
                (
                    typeof(SourceGenerator.Generator),
                    "Functions_ToUpper_Generated.g.cs",
                    SourceText.From(expectedSubNamespaceGenerated, Encoding.UTF8, SourceHashAlgorithm.Sha256)
                ),
                (
                    typeof(SourceGenerator.Generator),
                    "Program.g.cs",
                    SourceText.From(expectedProgramGenerated, Encoding.UTF8, SourceHashAlgorithm.Sha256)
                )
            },
            ExpectedDiagnostics =
            {
                new DiagnosticResult("AWSLambda0103", DiagnosticSeverity.Info).WithArguments("Functions_ToUpper_Generated.g.cs", expectedSubNamespaceGenerated),
                new DiagnosticResult("AWSLambda0103", DiagnosticSeverity.Info).WithArguments($"TestExecutableServerlessApp{Path.DirectorySeparatorChar}serverless.template", expectedTemplateContent),
            },
            ReferenceAssemblies = ReferenceAssemblies.Net.Net60
        }
    };

    foreach (var file in Directory.GetFiles(
                    Path.Combine("Amazon.Lambda.RuntimeSupport"),
                    "*.cs", SearchOption.AllDirectories))
    {
        var content = await File.ReadAllTextAsync(file);

        // Don't include RuntimeSupport's entry point.
        if (file.EndsWith("Program.cs") && content.Contains("Task Main(string[] args)"))
            continue;

        test.TestState.Sources.Add((file, content));
    }

    await test.RunAsync();
}
```